docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,214 @@
+User Customization & Persistence
+================================
+
+Context
+-------
+
+.. todo::
+
+   This section will be handled by the requirements PR.
+
+
+Design Choices
+--------------
+
+:term:`ConfigMap` KV(Key-Value) store is chosen as a unified data access and
+storage media for user editable configurations in a MetalK8s cluster based on
+the above requirements for the following reasons:
+
+* The limited number of read/write operations even for multi-node setups means
+  that concurrency issues on CRUD operations will not lag even if we do not
+  implement cross-app locking.
+* Ability to support CRUD operations on ConfigMap's with CLI and UI easily
```

Comment:
Why this focus on CRUD? I'm actually thinking, we only want the users to perform **U**pdate operations on those objects, since MetalK8s is responsible for **C**reating and **D**eleting those objects - so the fact that all CRUD ops are supported could be a problem (luckily, we can simply have a `Role` which only allows `patch` on the right objects).

Reply:
ok